packages/components/src/internal/components/picklist/actions.ts

```diff
@@ -248,35 +248,33 @@ export function getSelectedPicklistSamples(
     });
 }
 
-export function getSamplesNotInList(listName: string, selectionKey?: string, sampleIds?: string[]): Promise<string[]> {
-    return new Promise((resolve, reject) => {
-        getPicklistSamples(listName)
-            .then(existingSamples => {
-                if (sampleIds) {
-                    resolve(sampleIds.filter(id => !existingSamples.has(id.toString())));
-                } else if (selectionKey) {
-                    getSelected(selectionKey).then(response => {
-                        resolve(response.selected.filter(id => !existingSamples.has(id.toString())));
-                    });
-                } else {
-                    resolve([]);
-                }
-            })
-            .catch(reason => {
-                console.error(reason);
-                reject(reason);
-            });
+export function getSamplesNotInList(listName: string, selectionKey?: string, useSnapshotSelection?: boolean, sampleIds?: string[]): Promise<string[]> {
```

Comment:
nit. Nice first step to refactoring to async. Since `getPicklistSamples` and `getSelected` are already logging any errors and we're then just rethrowing the error I'd recommend removing the wrapping `Promise` and try/catch.

```ts
export async function getSamplesNotInList(
    listName: string,
    selectionKey?: string,
    useSnapshotSelection?: boolean,
    sampleIds?: string[]
): Promise<string[]> {
    const existingSamples = await getPicklistSamples(listName);

    if (sampleIds) {
        return sampleIds.filter(id => !existingSamples.has(id.toString()));
    }

    if (selectionKey) {
        const response = await getSelected(selectionKey, useSnapshotSelection);
        return response.selected.filter(id => !existingSamples.has(id.toString()));
    }

    return [];
}
```

Reply:
Good suggestion. Almost just too late, but I'll take it.